libc/src/sys/mman/linux/shm_open.cpp

```diff
@@ -0,0 +1,26 @@
+//===---------- Linux implementation of the shm_open function -------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include "src/sys/mman/shm_open.h"
+#include "src/fcntl/open.h"
+#include "src/sys/mman/linux/shm_common.h"
+#include <asm/fcntl.h>
+
+namespace LIBC_NAMESPACE {
+
+static constexpr int DEFAULT_OFLAGS = O_NOFOLLOW | O_CLOEXEC | O_NONBLOCK;
+
+LLVM_LIBC_FUNCTION(int, shm_open, (const char *name, int oflags, mode_t mode)) {
+  auto buffer = get_shm_name(name);
+  if (!buffer.has_value()) {
+    return -1;
+  }
```

Comment:
Small style nits. Please don't use auto here, and avoid `{}` for single statement conditionals/control flow.